src/components/IGVFileSelector.js

```diff
@@ -13,27 +12,80 @@ const getStrings = (v) => {
   return Utils.cond([_.isString(v), () => [v]], [!!v?.items, () => _.flatMap(getStrings, v.items)], () => []);
 };
 
-export const getValidIgvFiles = (values) => {
-  return _.flatMap((value) => {
-    const possibleFile = /(.+)\.([^.]+)$/.exec(value);
+const splitExtension = (fileUrl) => {
+  const extensionDelimiterIndex = fileUrl.lastIndexOf('.');
+  const base = fileUrl.slice(0, extensionDelimiterIndex);
+  const extension = fileUrl.slice(extensionDelimiterIndex + 1);
+  return [base, extension];
+};
 
-    if (_.isEmpty(parseGsUri(value)) || !possibleFile) {
-      return [];
-    }
+const isTdrUrl = (fileUrl) => {
+  const parts = fileUrl.split('/');
+  return (
+    parts.length === 6 &&
```

Comment:
Is it possible to make this check resilient to TDR adding additional components to our URL beyond what's presently there?

For backwards compatibility, we would not remove or reorder any existing components.

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4084/commits/28c1b2efa6aa6edc743f9bfad6126644b5e86730.